### PR TITLE
feature: user documents are uploaded to the server

### PR DESCRIPTION
**Pull request by cheedinmar**

### CODE REVIEW

File: components/Upload.tsx [hunk 1]
```diff
@@ -1,31 +1,96 @@
...
+          method: "post",
+          body: formData,
+          headers: {
+            Authorization: `Bearer ${authData.token}`,
+            "Content-Type": "multipart/form-data; ",
+          },
+        }
+      );
+      let responseJson = await res.json();
+      console.log(responseJson)
```
Comment (teebarg) [inline components/Upload.tsx]: remove log

Comment (cheedinmar) [reply components/Upload.tsx]: ok

File: screens/UploadDocument.tsx [hunk 2]
```diff
@@ -101,9 +85,12 @@ export default function Dashboard({ navigation, route }: Props) {
         url: `/auth/user`,
         headers: { Authorization: `Bearer ${authData.token}` },
       });
-      const user = response.data.data[0].attributes;
-      setUser(user);
-      setOnBoarded(user?.on_boarded);
+      const user = response.data.data[0];
+      setUser(user.attributes);
+      const upload = Object.values(user.included.verification).every(
+        (val) => val === true
```
Comment (teebarg) [inline screens/UploadDocument.tsx]: this is weird, what other value could val be?

File: components/Upload.tsx [hunk 3]
```diff
@@ -1,31 +1,97 @@
...
-       if (!result.cancelled) {
-         setImage(result.uri)
-       }
-     };
+    // Upload the image using the fetch and FormData APIs
+    let formData = new FormData();
+    formData.append("document", { uri: localUri, type:type, name: filename });
+    formData.append("type", props.type)
+      let res = await fetch(
+        "https://altara-customer-play-api.herokuapp.com/api/v1/document/upload",
```
Comment (teebarg) [inline components/Upload.tsx]: can we put the base api in an ENV, so we can swap when building for production?

File: screens/UploadDocument.tsx (hunk 2 above)
Comment (cheedinmar) [reply screens/UploadDocument.tsx]: for new customer's values are false, until the customer uploads documents. The function is a truthy check  if any document is not uploaded it returns false and true only when all documents are true
image

File: components/Upload.tsx (hunk 3 above)
Comment (cheedinmar) [reply components/Upload.tsx]: done

File: screens/UploadDocument.tsx (hunk 2 above)
Comment (teebarg) [reply screens/UploadDocument.tsx]: will returning just `val` not do the same thing?

File: navigation/index.tsx [hunk 4]
```diff
@@ -175,38 +176,41 @@ function DrawerNavigator({ route, navigation }) {
 			});
 			const user = response.data.data[0];
 			setUser(user);
+			 const upload = Object.values(user?.included?.verification || {}).every(val => val === true);
```
Comment (teebarg) [inline navigation/index.tsx]: Same here

**Review by teebarg — APPROVED** [review]

**Pull request merged**